tests/Routing/RoutingRouteTest.php

```diff
@@ -309,24 +309,6 @@ public function testClassesCanBeInjectedIntoRoutes()
         unset($_SERVER['__test.route_inject']);
     }
 
-    public function testClassesAndVariablesCanBeInjectedIntoRoutes()
```

Comment:
To transform a route method dependency we first check if the class already exists in the [parameters passed to the route](https://github.com/laravel/framework/blob/5.4/src/Illuminate/Routing/RouteDependencyResolverTrait.php#L75) if so then we assume it's a Model so we pass it as is.

The given test contains two type-hinted stdClass arguments, checking for the second stdClass we'll find that it's already in parameters when the first stdClass was resolved, as a result we won't resolve it and we pass the default value instead.

That test used to work before because we were building a brand new array of results instead of mutating the `$parameters` array, and thus when we check for the second stdClass in `$parameters` it won't exist since the first resolved one was stored in the `$results` array not the `$parameters` array.

For example before this PR you could have `function show(Request $request, $id, Request $sameRequest)` and you get two arguments with a Request instance, which isn't considered a breaking change because it actually doesn't make sense to resolve the same method dependency twice.